pkg/discovery/mesos_leader.go

```diff
@@ -0,0 +1,243 @@
+package discovery
+
+import (
+	"github.com/turbonomic/mesosturbo/pkg/conf"
+	master "github.com/turbonomic/mesosturbo/pkg/masterapi"
+	"fmt"
+	"strings"
+	"github.com/golang/glog"
+	"github.com/turbonomic/mesosturbo/pkg/data"
+	"strconv"
+)
+
+
+type MASTER_IP_PORT string
+// Structure representing the Master acts as the leader in the cluster
+type MesosLeader struct {
+	// Mesos Target configuration provided to the mesosturbo service
+	targetConf       *conf.MesosTargetConf
+	// Map of master IP:Port and the Rest API Client for that master
+	restApiClientMap map[MASTER_IP_PORT]master.MasterRestClient
```

Comment:
Is it necessary to keep all the restApiClient?
If the leader changed, we can create a restApiClient; If not, nothing to do.